### DISCUSSION TITLE
Working with subgraph

### DISCUSSION THREAD

**Discussion by AlessandroGussoni** [pyg-team/pytorch_geometric#8140] [Q&A]
Hi all! are there any available resources about dealing with large graph and how to work with subgraph? In particular i would be interested in saving a large Data object and at each epoch load use a custom function to load in memory a subgraph to train on it. Ideally at each epoch my custom function would select a new random sample of the whole graph. Any idea on how to implement this?
Many thanks :)

**Comment by akihironitta** [comment]
I'd suggest [Heterogeneous Graph Learning - Heterogeneous Graph Samplers](https://pytorch-geometric.readthedocs.io/en/latest/tutorial/heterogeneous.html#heterogeneous-graph-samplers) and the examples using `NeighborLoader` in [`examples/`](https://github.com/search?q=repo%3Apyg-team%2Fpytorch_geometric+NeighborLoader+path%3A%2F%5Eexamples%5C%2F%2F&type=code).

**Reply by AlessandroGussoni** [reply]
Hi, thanks for the reply! Those look really helpful, i have just a couple of followup questions. In each Data object i have a graph, a vectore of different graph features and a vector of y for each of the graph config. 
How can i load specific nodes plus some random ones? (i.e. i have 100 nodes, i would like to always load node at index (10, 35) and some 10 neighbors for each of the two nodes)
Each of my Data object as lots of different graph features (saved in a matrix) with related labels (saved in a vector), how can i load from disk a random sample of them? 

Thanks again for the answer!

**Reply by rusty1s** [reply]
Do you have multiple large graphs or a single one? In the end, `NeighborLoader` is designed to operate on a single large graph, where the `input_nodes` specify the nodes you want to make a prediction for (e.g., 10 and 35 in your case), and `num_neighbors=[10]` would specify that you sample 10 neighbors from it.
